src/window.rs

```diff
@@ -56,6 +56,48 @@ struct WindowWrapper {
     fullscreen: bool,
 }
 
+pub fn window_geometry() -> Result<(u64, u64), String> {
+    let prefix = "--geometry=";
+    let arg = std::env::args()
+        .filter(|arg| arg.starts_with(prefix))
+        .next();
+    match arg {
+        Some(sizes_str) => {
+            let input = sizes_str.split_at(prefix.len()).1;
```

Comment:
This could probably be simplified with something like https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=827c58281dd17cf14eb720675b255f3c